### ISSUE TITLE
Values are not updated if you change initialValues

### ISSUE THREAD

**Issue by ghalex** [BenMagyar/form-hooks#7]
If you set `initialValues` from `props` and props update the `values` don't update.

I have made this example to demonstrate the bug:
https://codesandbox.io/s/16p2053qq

The solution I think after checking the code will be to add this:

```
  useEffect(() => {
    setValues(initialValues);
  }, [initialValues]);
```

so if the initial values change the values should change to.

**Comment by BenMagyar** [comment]
Hmm, I didn't think about that case (and honestly haven't updated this since 16.7-alpha 😳). I'll update it this weekend, it does make sense to me that switching the `initialValues` set re-sets the form 👍.

**Comment by BenMagyar** [comment]
@ghalex I just released 0.5.0 that has support for this. You can now list a collection of dependencies from the `FormHookOptions<Values>` object that is used during form creation. The below example will re-initialize on `initialValues` changing.

```ts
useForm({
  initialValues: { first: 'John' },
  ...
}, options => [options.initialValues])
```

See https://codesandbox.io/s/zl26jkkky4 for a working example and [added secion](https://github.com/BenMagyar/form-hooks#formhookdependenciesvalues---form-reinitialization) in documentation for more information. Thanks!

**Comment by ghalex** [comment]
Thanks a lot for this @BenMagyar. I think you did a very good job with this library.
